### PR TITLE
New: Subtotal bard adapter.

### PR DESCRIPTION
**Pull request by cythrawll**
## Description

Port of subtotal adapter from alpha branch

## Proposed Changes

- Since reqv2 uses CIDs rather than dimensions, we have to normalize to dimension names in the adapter.

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/data/addon/adapters/facts/bard.ts [hunk 1]
```diff
@@ -266,6 +266,27 @@ export default class BardFactsAdapter extends EmberObject implements NaviFactAda
...
+  /**
+   * Builds rollup query params
+   * @param {RequestV1} request
+   * @return {String}
+   */
+  _buildRollupParam(request: RequestV2): string {
+    const columns = request.columns;
+    const rollupColumns = request.rollup?.columns || [];
+    return rollupColumns.length
+      ? array(
```
Comment (kevinhinterlong) [inline packages/data/addon/adapters/facts/bard.ts:278]: Can we avoid ember array and make cid -> field map
```js
const cidToField = request.columns.reduce((cidToField, column) => {
  const field = column && isDateTime(column) ? 'dateTime' : column?.field;
  cidToField.set(column.cid, field);
  return cidToField
}, new Map());
```

**Review by kevinhinterlong — COMMENTED** [review]
small stuff

File: packages/data/addon/adapters/facts/bard.ts [hunk 2]
```diff
@@ -266,6 +266,24 @@ export default class BardFactsAdapter extends EmberObject implements NaviFactAda
...
+   * Builds rollup query params
+   * @param {RequestV1} request
+   * @return {String}
+   */
+  _buildRollupParam(request: RequestV2): string {
+    const columns = request.columns;
+    const rollupColumnCids = request.rollup?.columns || [];
+    return [
+      ...columns.reduce((colSet, requestColumn) => {
+        if (rollupColumnCids.includes(requestColumn.cid ?? '')) {
```
Comment (daniellevinson) [inline packages/data/addon/adapters/facts/bard.ts:279]: can `rollupColumnCids` actually include `''` or is it just to return false?

File: packages/data/addon/adapters/facts/bard.ts [hunk 3]
```diff
@@ -266,6 +266,24 @@ export default class BardFactsAdapter extends EmberObject implements NaviFactAda
     return undefined;
   }
 
+  /**
+   * Builds rollup query params
+   * @param {RequestV1} request
```
Comment (daniellevinson) [inline packages/data/addon/adapters/facts/bard.ts:271]:
```suggestion
   * @param {RequestV2} request
```

**Review by daniellevinson — APPROVED** [review]

File: packages/data/addon/adapters/facts/bard.ts (hunk 2 above)
Comment (cythrawll) [reply packages/data/addon/adapters/facts/bard.ts:279]: just to return false.

**Pull request merged**